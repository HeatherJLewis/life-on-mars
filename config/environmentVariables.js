const dotenv = require("dotenv");
const environment = dotenv.config();
if (environment.error) {
  throw environment.error;
}
const PORT = process.env.PORT;
const NASA_API_KEY = process.env.NASA_API_KEY;

module.exports = { PORT, NASA_API_KEY };
