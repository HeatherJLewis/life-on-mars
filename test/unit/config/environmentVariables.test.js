describe("environment variables", () => {
  let dotenv;
  let originalProcessEnv;

  beforeEach(() => {
    dotenv = require("dotenv");
    jest.mock("dotenv");
    originalProcessEnv = process.env;
  });

  afterEach(() => {
    jest.resetModules();
    process.env = originalProcessEnv;
  });

  it("should export an object with a property of PORT", () => {
    dotenv.config.mockReturnValue({});
    process.env = {
      PORT: "somePort",
      NASA_API_KEY: "someApiKey",
    };

    const exportValue = require("../../../config/environmentVariables");

    expect(exportValue).toMatchSnapshot();
  });

  it("should call config", () => {
    dotenv.config.mockReturnValue({});
    require("../../../config/environmentVariables");

    expect(dotenv.config).toHaveBeenCalled();
  });

  it("should throw an error if .config returns an error", () => {
    dotenv.config.mockReturnValue({
      error: new Error(),
    });

    expect(() => {
      require("../../../config/environmentVariables");
    }).toThrowError();
  });
});
