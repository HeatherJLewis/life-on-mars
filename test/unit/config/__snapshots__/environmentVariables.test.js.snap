// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`environment variables should export an object with a property of PORT 1`] = `
Object {
  "NASA_API_KEY": "someApiKey",
  "PORT": "somePort",
}
`;
